dj/api/nodes.py

```diff
@@ -186,18 +206,16 @@ def read_nodes(*, session: Session = Depends(get_session)) -> List[NodeMetadata]
     """
     List the available nodes.
     """
-    nodes = session.exec(select(Node)).all()
+    nodes = session.exec(select(Node).options(joinedload(Node.current))).all()
     return nodes
 
 
-@router.get("/nodes/{name}/", response_model=NodeWithRevisions)
-def read_node(
-    name: str, *, session: Session = Depends(get_session)
-) -> NodeWithRevisions:
+@router.get("/nodes/{name}/", response_model=NodeMetadata)
+def read_node(name: str, *, session: Session = Depends(get_session)) -> NodeMetadata:
     """
-    List the specified node and include all revisions.
+    Show the active version of the specified node.
     """
-    statement = select(Node).where(Node.name == name)
+    statement = select(Node).where(Node.name == name).options(joinedload(Node.current))
```

Comment:
Does this `joinedload` mean the query filters to the node but then only returns the current node revision that's linked? This is pretty cool.